src/idpi/products/pot_vortic.py

```diff
@@ -0,0 +1,68 @@
+# Standard library
+import sys
+import logging
+
+# Third-party
+import numpy as np
+
+# First-party
+from idpi.operators.destagger import destagger
+from idpi.operators.pot_vortic import fpotvortic
+from idpi.operators.rho import f_rho_tot
+from idpi.operators.theta import ftheta
+from idpi.operators.total_diff import TotalDiff
+from idpi.operators.vertical_interpolation import interpolate_k2p
+from idpi.operators.vertical_interpolation import interpolate_k2theta
+from idpi.operators.vertical_reduction import integrate_k
+
+logging.basicConfig(stream=sys.stdout, level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+def _compute_pot_vortic(U, V, W, T, P, QV, QC, QI, HHL, theta):
```

Comment:
The goal of this function is to restrict the lifetime of `rho_tot` and `total_diff` to minimise the memory footprint, there's no guarantee that it will remain once we've moved to `dask.delayed` or some other computation framework.

Reply:
I think that the question of how the API should look like still needs to be discussed.

Reply:
yes, that is fine. But if this is not a public operator, the next product (there is more than one) that needs to compute pot_vortic will have to implement a copy of this private function.

Reply:
The API to compute the potential vorticity is the one that is found in the operators. I wouldn't multiply the number of ways to compute the same thing.